EduID/Flows/CreateEduID/ViewControllers/CreateEduIDRegistrationCheckViewController.swift

```diff
@@ -0,0 +1,109 @@
+import UIKit
+import TinyConstraints
+import Combine
+
+enum RegistrationCheckAlertMode {
+    case error, deactivate
+}
+
+class CreateEduIDRegistrationCheckViewController: CreateEduIDBaseViewController {
+    
+    let activityIndicator = UIActivityIndicatorView(style: .large)
+    let viewModel: CreateEduIDRegistrationCheckViewModel = .init()
+    private var cancellable : Set<AnyCancellable> = .init()
+    private var shouldShowNextScreen: Bool = true
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupUI()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        activityIndicator.startAnimating()
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        if AppAuthController.shared.isLoggedIn() {
+            Task {
+                await viewModel.checkForAnyExistingUser()
+                await MainActor.run {
+                    setupObservers()
```

Comment:
I would set up the observers first, and then start checking for an existing user